server/controllers/user.controllers.js

```diff
@@ -239,6 +240,18 @@ const UserController = {
       })
       .catch((error) => next(error));
   },
+
+  updateVendor(req, res, next) {
+    const updatedVendor = req.locals;
+    const vendorId = req.user._id;
+    updateVendor({ updatedVendor, vendorId })
+      .then((vendor) => {
+        res
+          .status(httpStatus.OK)
+          .json({ success: true, message: 'Vendor information updated', vendor });
```

Comment:
Let's return this as a user so that we can have something like `user.vendor` and not `vendor.vendor` to get vendor specific information. 
```suggestion
          .json({ success: true, message: 'Vendor information updated', user });
```

It will also make the tests more readable. Having 2 vendors can be confusing:
```js
expect(res.body.vendor.vendor.companyAddress).to.be.eql(data.companyAddress);
```

Please note that some tests might fail when you update this so ensure you update the tests too.